filters/openpolicyagent/opaauthorizerequest/opaauthorizerequest_test.go

```diff
@@ -57,6 +57,20 @@ func TestAuthorizeRequestFilter(t *testing.T) {
 			backendHeaders:    make(http.Header),
 			removeHeaders:     make(http.Header),
 		},
+		{
+			msg:               "Allow Requests with query parameters",
+			filterName:        "opaAuthorizeRequest",
+			bundleName:        "somebundle.tar.gz",
+			regoQuery:         "envoy/authz/allow",
+			requestPath:       "/allow-with-query?pass=yes&q2=v2",
```

Comment:
Shall we add a test case to cover, `/allow-with-query?q2=yes&q2=v2` case as well as for the behavior mentioned [here](https://www.openpolicyagent.org/docs/latest/envoy-primer/#example-input).

`/people?lang=en&id=1&id=2` should give `input.parsed_query.id == ["1", "2"]`

Reply:
recently we found a good special case, https://github.com/zalando/skipper/pull/3201 a testcase can also be good for this to see how it will be handled by opa`/allow-with-query?`

Reply:
> recently we found a good special case, #3201 a testcase can also be good for this to see how it will be handled by opa`/allow-with-query?`

Do you mean the test case for `/allow-with-query?` or `?` trailing in the query params?

Reply:
> Shall we add a test case to cover, `/allow-with-query?q2=yes&q2=v2` case as well as for the behavior mentioned [here](https://www.openpolicyagent.org/docs/latest/envoy-primer/#example-input) when same param ID has multiple values.
> 
> `/people?lang=en&id=1&id=2` should give `input.parsed_query.id == ["1", "2"]`

Modifed the same test case to include both scenarios

Reply:
> Do you mean the test case for `/allow-with-query?` or `?` trailing in the query params?

yes, I meant the trailing `?` with no query parameters.

Reply:
> > Do you mean the test case for `/allow-with-query?` or `?` trailing in the query params?
> 
> yes, I meant the trailing `?` with no query parameters.

I assumed this was the case and added a test case :)